### PR TITLE
Feature: websocket interface

### PR DESCRIPTION
**Pull request by jamesjwarren**
Configures socket.io to listen to thisissoon.fm socket server. Event handler's can be added by injecting `fmSocket` into a controller and calling `fmSocket.addListener(event, eventHandler)`. The eventHandler will be called with the data object passed with the event. This may include track data etc.

### CODE REVIEW

File: app/js/fm-socket/socket.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        /**
+         * @constructor
+         * @param   {Factory} socketFactory  factory for creating socket instances
+         * @returns {Object}  socket.io instance
+         */
+        function (socketFactory) {
+
+            return socketFactory({
+                prefix: "",
+                ioSocket: io.connect("http://localdocker:8080")
```
Comment (eopa) [inline app/js/fm-socket/socket.js]: This URL should be defined as an angular constant

**Comment by eopa** [comment]
@jamesjwarren Is there anyway you can create an example of using this?

**Comment by jamesjwarren** [comment]
This now includes an initialisation service which forwards the available socket events to angular event system i.e. `$scope.$on("fm:player:play", ...`. I've also added an example app using the socket in `js/fm-socket/example`

**Comment by eopa** [comment]
:+1: looks good

**Pull request merged**